pkg/resources/network/hostname_status.go

```diff
@@ -0,0 +1,94 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+)
+
+// HostnameStatusType is type of HostnameStatus resource.
+const HostnameStatusType = resource.Type("HostnameStatuses.net.talos.dev")
+
+// HostnameStatus resource holds node hostname.
+type HostnameStatus struct {
+	md   resource.Metadata
+	spec HostnameStatusSpec
+}
+
+// HostnameStatusSpec describes node nostname.
+type HostnameStatusSpec struct {
+	Hostname   string `yaml:"hostname"`
+	Domainname string `yaml:"domainname"`
+}
+
+// FQDN returns the fully-qualified domain name.
+func (spec *HostnameStatusSpec) FQDN() string {
+	if spec.Domainname == "" {
+		return spec.Hostname
+	}
+
+	return spec.Hostname + "." + spec.Domainname
+}
+
+// NewHostnameStatus initializes a SecretsStatus resource.
```

Comment:
```suggestion
// NewHostnameStatus initializes a HostnameStatus resource.
```

And it looks like all other resources have the same copy-paste error ;)

Reply:
thanks, fixed all of them!